src/Apps/cudaPlanesweepMAPTk.cpp

```diff
@@ -400,9 +397,14 @@ int main(int argc, char* argv[])
             // Compute near and far range.
             double nearZ,farZ;
             std::vector<float> zs;
+            const Eigen::Matrix3d& K = it->second.getK();
+            // Assume the image size is twice the principal point
+            const int imageWidth=static_cast<int>(K(0,2) * 2);
+            const int imageHeight=static_cast<int>(K(1,2) * 2);
+            std::cout<< "Image " << it->first << " is assumed to be " << imageWidth << "x"<< imageHeight <<std::endl;
             for(unsigned int i=0; i<points.size(); i++) {
                 Eigen::Vector3d x = (it->second.getR()*points[i] + it->second.getT());
-                Eigen::Vector3d px= 1.0/x[2]*(it->second.getK()*x);
+                Eigen::Vector3d px= 1.0/x[2]*(K * x);
                 if(x[2]>0 && px[0]>0 && px[0]<imageWidth && px[1]>0 && px[1]<imageHeight)
```

Comment:
This line is supposed to select 3D point that are in front of the camera and projects within image. This would be affected by taking 2*principal point as the image size.